src/components/profile/EditAvatar.js

```diff
@@ -31,9 +32,16 @@ const EditAvatar = ({ screenProps, theme }) => {
       showErrorDialog('We could not capture all your beauty. Please try again.')
     })
 
+    navigation.navigate('ViewAvatar')
     setSaving(false)
   }
 
+  useEffect(() => {
+    if (!avatarRef.current) {
```

Comment:
why you created ref for current avatar if you have an state var for this ?
You could just do:

```
 const [avatar, setAvatar] = useState(profile.avatar)

useEffect(() => setAvatar(profile.avatar), [profile])

<ImageCropper image={avatar} onChange={handleAvatarChange} />
```

Reply:
```javascript
const EditAvatar = ({ theme, navigation }) => {
  const store = GDStore.useStore()
  const [showErrorDialog] = useErrorDialog()
  
  const user = useWrappedUserStorage()
  const profile = store.get('profile')
  
  const [isDirty, markAsDirty] = useState(false)
  const [processing, setProcessing] = useState(false)

  const [avatar, setAvatar] = useState(profile.avatar)
  const croppedRef = useRef(avatar)

  const updateAvatar = useCallback(async () => {
    setProcessing(true)

    try {      
      await user.setAvatar(avatar)
    } catch (exception) {
      const { message } = exception

      log.error('saving image failed:', message, exception)
      showErrorDialog('We could not capture all your beauty. Please try again.')
    } finally {
      setProcessing(false)
    }

    navigation.navigate('ViewAvatar')    
  }, [navigation, markAsDirty, setProcessing, showErrorDialog, user])

  const onCropped = useCallback(cropped => {
    croppedRef.current = cropped
    markAsDirty(true)
  }, [markAsDirty])

  useEffect(() => {    
    if (!processing) {
      const { avatar } = profile

      setAvatar(avatar)
      markAsDirty(false)

      croppedRef.current = avatar
    }
  }, [profile])

  return (
    <Wrapper>
      <Section style={styles.section}>
        <Section.Row>
          <ImageCropper image={avatar} onChange={onCropped} />
        </Section.Row>
        <Section.Stack justifyContent="flex-end" grow>
          <CustomButton
            disabled={!isDirty || processing}
            loading={processing}
            onPress={updateAvatar}
            color={theme.colors.darkBlue}
          >
            Save
          </CustomButton>
        </Section.Stack>
      </Section>
    </Wrapper>
  )
}
```

Reply:
done